src/harness/reference_models/move_list_calc/move_list.py

```diff
@@ -0,0 +1,435 @@
+                    to the CBSDs
+
+    Returns:
+        an array contains index of CBSD inside the main beam
+    """
+
+    idx = []
+
+    # Compute off-axis angle between the axis of the main-beam of the incumbent
+    # receiver and the line between the protection point and each CBSD
+    for i in range(len(bearing)):
+        theta = ant_az - bearing[i]
+        if theta > 180:
+            theta -= 360
+        elif theta < -180:
+           theta += 360
+
+        # Check if the CBSD is inside or outside the antenna main-beam
+        if (theta >= (-1) * RADAR_BEAMWIDTH/2.0) and (theta <= RADAR_BEAMWIDTH/2.0):
+            idx.append(i)
+
+    return np.array(idx)	
+
+
+def find_nc(I, bearing, t):
+
+    """
+    Return the index (nc) of the grant in the ordered list of grants such that
+    the 95th percentile of the interference from the first nc grants is below the
+    threshold for all azimuths of the receiver antenna.
+
+    Inputs:
+        I:      2D array of interference contributions (dBm/10 MHz); columns
+                correspond to grants, and rows correspond to Monte Carlo iterations
+        t:      95th-percentile protection threshold (dBm/10 MHz)
+
+    Returns:
+        nc:     index nc that defines the move list to be {G_nc+1, G_nc+2, ..., G_Nc}
+    """
+
+    # Create array of incumbent antenna azimuth angles (degrees).
+    azimuths = np.arange(0.0, 360.0, RADAR_BEAMWIDTH/2.0)
+
+    # Initialize nc to Nc.
+    Nc = I.shape[1]
+    nc = Nc
+
+    # Convert protection threshold, interference matrix, and antenna
+    # discrimination to linear units.
+    t_mW = np.power(10.0, t/10.0)
+    I_mW = np.power(10.0, I/10.0)
+    G_discrim = np.power(10.0, INTRF_ATTEN_OUTSIDE_MAIN_BEAM/10.0)
+
+    # Loop through every azimuth angle.
+    for a in azimuths:
+        # Calculate interference contributions at output of receiver antenna.
+        G = np.full((1,Nc),G_discrim)
+
+        idx = grantsInMainBeam(a,bearing)
+        if idx.size:
+            G[:,idx] = 1.0
+        IG = I_mW*G
+
+        # Compute the 95th percentile of the aggregate interference, and remove
+        # grants until the protection threshold is met or all grants are moved.
+        A95 = np.percentile(IG.sum(1), PROTECTION_PERCENTILE, interpolation='lower')
+        while (A95 > t_mW):
+            I_mW[:, nc - 1] = 0.0  # zero-out the nc-th column of I_mW
+            IG[:, nc - 1] = 0.0    # zero-out the nc-th column of IG
+            nc = nc - 1
+            if nc == 0:
+                return 0
+            A95 = np.percentile(IG.sum(1), PROTECTION_PERCENTILE, interpolation='lower')
+
+    return nc
+
+
+def findMoveList(protection_points, low_freq, high_freq, threshold, num_iter,
+                 registration_request, grant_request, exclusion_zone):
+
+    """
+    Main routine to find CBSD indices on the move-list.
+
+    Inputs:
+        protection_points:  a list of protection points
```

Comment:
For the top level function, can we describe better the format of inputs and outputs.
For example the protection points currently assume being a list of dict of 'latitude' and 'longitude', etc..